src/main/scala/codacy/pylint/Pylint.scala

```diff
@@ -36,24 +45,90 @@ object Pylint extends Tool {
 
   private val msgTemplate = "{path}###{line}###{msg}###{msg_id}"
 
-  private def commandFor(path: Path, conf: Option[Seq[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[Seq[String]] = {
+  private def collectFiles(files: Option[Set[Path]], path: Path) = {
+    files.collect { case files if files.nonEmpty => files.map(_.toString) }.getOrElse {
+      //if files is empty, let the classification script to find them.
+      Seq(path.toString)
+    }
+  }
+
+  def printToFile(file: java.io.File, text: String) = {
+      val printer = new java.io.PrintWriter(file)
+      try { printer.write(text) } finally { printer.close() }
+  }
+
+  def generateClassification(files: Iterable[String]) = {
+
+    val tmp = File.createTempFile("pylint", "")
+    val script_args = files.mkString("###")
+    val script = s"""
+         |import os
+         |import sys
+         |import ast
+         |current = sys.version_info[0]
+         |other = 2 if current == 3 else 3
+         |def _classify_file(path):
+         |    try:
+         |        with open(path, 'r') as stream:
+         |            try:
+         |                ast.parse(stream.read())
+         |            except (ValueError, TypeError, UnicodeError):
+         |                # Assume it's the current interpreter.
+         |                return current
+         |            except SyntaxError:
+         |                # the other version or an actual syntax error on current interpreter
+         |                return other
+         |            else:
+         |                return current
+         |    except Exception:
+         |        # Shouldn't happen, but if it does, just assume there's
+         |        # something inherently wrong with the file.
+         |        return current
+         |def classify_file(path):
+         |    interpreter = _classify_file(path)
+         |    return path + "###" + str(interpreter)
+         |def flatten_files(folder):
+         |    for _, _, files in os.walk(folder):
+         |        for file in files:
+         |            if file.endswith(".py"):
+         |                yield os.path.join(folder, file)
+         |def walk_items(items):
+         |    for item in items:
+         |        if os.path.isfile(item): yield item
+         |        elif os.path.isdir(item):
+         |            for file in flatten_files(item): yield file
+         |def classify(items):
+         |    for file in walk_items(items):
+         |        print(classify_file(file))
+         |items = filter(None, sys.argv[1].split("###"))
+         |classify(items)    
+       """.stripMargin
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://www.codacy.com/app/Codacy/codacy-pylint/file/1728033956/issues/source?bid=2829823&fileBranchId=3054541#l106)